frontend/packages/core/src/dialog.tsx

```diff
@@ -64,19 +64,29 @@ export interface DialogProps extends Pick<MuiDialogProps, "open"> {
     | React.ReactElement<DialogContentProps>[]
     | React.ReactElement<DialogActionsProps>
     | React.ReactElement<DialogActionsProps>[];
+  onClose?: () => void;
```

Comment:
actually this completely makes sense since we already require state from the parent component. Let's change this to be required, remove the conditional and update the references.

Reply:
done! can you please take a look and give me your thoughts on the `DialogCloseReasons` type? The Material Dialog unfortunately doesn't export the close reasons and instead just lists them in the JSdoc comment: 
![image](https://user-images.githubusercontent.com/1319951/105552841-bb8dad80-5cc1-11eb-95af-24e56822824a.png)